src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -135,55 +137,163 @@ void CommissionerDiscoveryController::Ok()
     }
     client->SetUDCClientProcessingState(UDCClientProcessingState::kObtainingOnboardingPayload);
 
-    if (mPasscodeService != nullptr)
+    if (mPasscodeService == nullptr)
     {
-        char rotatingIdString[chip::Dnssd::kMaxRotatingIdLen * 2 + 1] = "";
-        Encoding::BytesToUppercaseHexString(client->GetRotatingId(), client->GetRotatingIdLength(), rotatingIdString,
-                                            sizeof(rotatingIdString));
-        // Encoding::BytesToUppercaseHexString(client->GetRotatingId(), chip::Dnssd::kMaxRotatingIdLen, rotatingIdString,
-        //                                     sizeof(rotatingIdString));
+        HandleContentAppPasscodeResponse(0);
+        return;
+    }
+
+    char rotatingIdString[chip::Dnssd::kMaxRotatingIdLen * 2 + 1] = "";
+    Encoding::BytesToUppercaseHexString(client->GetRotatingId(), client->GetRotatingIdLength(), rotatingIdString,
+                                        sizeof(rotatingIdString));
+    CharSpan rotatingIdSpan = chip::CharSpan(rotatingIdString, sizeof(rotatingIdString));
 
-        CharSpan rotatingIdSpan = chip::CharSpan(rotatingIdString, sizeof(rotatingIdString));
-        uint32_t passcode       = 0;
-        uint8_t targetAppCount  = client->GetNumTargetAppInfos();
-        if (targetAppCount > 0)
+    uint8_t targetAppCount = client->GetNumTargetAppInfos();
+    if (targetAppCount > 0)
+    {
+        for (uint8_t i = 0; i < targetAppCount; i++)
         {
-            bool hasTargetApp = false;
-            for (uint8_t i = 0; i < targetAppCount; i++)
+            TargetAppInfo info;
+            if (client->GetTargetAppInfo(i, info))
             {
-                TargetAppInfo info;
-                if (client->GetTargetAppInfo(i, info))
+                mPasscodeService->LookupTargetContentApp(client->GetVendorId(), client->GetProductId(), rotatingIdSpan, info);
+            }
+        }
+        return;
+    }
+
+    mPasscodeService->FetchCommissionPasscodeFromContentApp(client->GetVendorId(), client->GetProductId(), rotatingIdSpan);
+    return;
+}
+
+void CommissionerDiscoveryController::HandleTargetContentAppCheck(TargetAppInfo target, uint32_t passcode)
+{
+    chip::DeviceLayer::StackLock lock;
```

Comment:
Again, you should probably not be using StackLock.

Reply:
removed the StackLock calls